signinapp/model.py

```diff
@@ -164,13 +188,57 @@ def make(name: str, password: str, role: Role, preferred_name: str = None,
                     subteam=subteam,
                     approved=approved)
 
+    @staticmethod
+    def make_guardian(student: User, name: str, phone_number: str, email: str, preference: int = 1):
+        guardian_limted = Role.from_name("guardian_limited")
+        guardian = User(name=name,
+                        code=mk_hash(name),
+                        role_id=guardian_limted.id,
+                        phone_number=normalize_phone_number_for_storage(
+                            phone_number),
+                        email=email
+                        )
+        db.session.add(guardian)
+        db.session.flush()
+        guardian.guardian = Guardian.add_guardian(
+            student=student, parent_id=guardian.id, preference=preference)
+        return guardian
+
     @ staticmethod
     def get_canonical(name) -> User | None:
         ' Look up user by name '
         code = mk_hash(name)
         return User.query.filter_by(code=code).one_or_none()
 
 
+class Guardian(db.Model):
+    """
+    This table is a bit strange as it has a one to one link with a User (Parent) as well as
+    Many to Many links with User (Children).
+
+    """
+    __tablename__ = "guardians"
+    id = db.Column(db.Integer, primary_key=True)
+    user_id = db.Column(db.Integer, db.ForeignKey("users.id"))
+    preference = db.Column(db.Integer)
```

Comment:
What is `preference`?

Reply:
Basically order to contact the guardian in, more useful if we allow more than 2 guardians to be added (needs a better UI). 
Would `contact_order` work? Or can you think of a better term?

Reply:
`contact_order` sounds great